### PR TITLE
Adding Getting Started page

### PR DESCRIPTION
**Pull request by dhmlau**
Fixes https://github.com/strongloop/v4.loopback.io/issues/22

### CODE REVIEW

**Comment by raymondfeng** [comment]
Should it be merged into `gh-pages` or `master`?

**Comment by dhmlau** [comment]
we're using the `master` branch

File: getting-started.html [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+        </div>
+    </nav>
+    <!-- Header -->
+    <div class="gs-masthead">
+        <div class="container-fluid"></div>
+        <div style="height:80px"></div>
+        <h2>Ready to build with LoopBack?</h2>
+        <br>
+        <h5 style="font-weight: normal">Let's get you eqipped</h5>
```
Comment (virkt25) [inline getting-started.html]: Need a `.` at the end of the sentence.

File: getting-started.html [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                    <a class="github-button" href="https://github.com/strongloop/loopback-next" data-icon="octicon-star" data-show-count="true"
+                        aria-label="Star strongloop/loopback-next on GitHub">Star</a>
+                </li>
+
+            </ul>
+
+        </div>
+    </nav>
+    <!-- Header -->
+    <div class="gs-masthead">
```
Comment (virkt25) [inline getting-started.html]: This background image needs to be muted -- spec seems to have an opacity overlay from top to bottom which gives better readability of the text on this (especially the smaller second line). Please implement that!

File: getting-started.html [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        <div class="container-fluid"></div>
+        <div style="height:80px"></div>
```
Comment (virkt25) [inline getting-started.html]: Inline styles should be avoided as much as possible.

File: getting-started.html (hunk 1 above)
Comment (shimks) [reply getting-started.html]: `eqipped`?
did you mean equipped?

File: css/getting-started.css [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    padding-left: 14px;
+    padding-right: 14px;
+    float: left;
+}
+
+code {
+    color: black;
+    background-color: var(--color-white-xd);
+    padding: 5px;
+}
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply css/getting-started.css]: let me try to set up linting or get some help from you on setting this up. thanks.

File: getting-started.html [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+                <div class="col-md-12">
+                    <button class="steptag text-uppercase">Step 1</button>
+                </div>
+                <div class="col-md-6">
+                    <h3 class="feature-heading">Have you installed Node.js?</h3>
+                    <p class="p large">Before you install LoopBack, make sure to download and install Node.js (version 8.x.x or higher), a JavaScript
+                        runtime. </p>
+                </div>
+                <div class="col-md-6">
+                    <a class="btn btn-secondary" style="padding-left:60px; padding-right:60px" role="button" href="https://nodejs.org/en/download/">Install Node.js</a>
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: for some reason, the border color is changed when it's in hover.  will fix that. thx

File: getting-started.html [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+                    <button class="steptag text-uppercase">Step 5</button>
+                </div>
+                <div class="col-md-6">
+                    <h3 class="feature-heading">Add your own controller</h3>
+                    <p class="p large">Now that we have a basic project created, it’s time to add our own controller. Let’s add a simple “Hello
+                        World” controller as follows.
+                    </p>
+                </div>
+                <div class="col-md-6">
+                    <p class="code-snippet" style="top:20%">
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: will fix this and other occurrences. thx

File: getting-started.html [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+            </div>
+        </div>
+ 
+        <hr class="hairline">
+
+        <!-- answer the prompt subsection -->
+        <div class="container">
+            <div class="row text-center">
+                <div class="col-md-6">
+                    <p class="p large" style="position:relative; top:50%">Answer the prompts as follows.
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: not sure if i know what you mean.  you mean it's not aligned vertically?

File: getting-started.html (hunk 2 above)
Comment (dhmlau) [reply getting-started.html]: I'll change the opacity from 0.4 to 0.2 in the new version.  Hope it's better.

Comment (dhmlau) [reply getting-started.html]: got it

File: getting-started.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: i've made it now so that the left text is somewhat in the middle relatively to the right column.

**Comment by dhmlau** [comment]
@virkt25 @shimks , I've made a few more changes according to your feedback (@shimks gave his verbally):
- spacing below the step title
- hairline under the navbar
- gradient opacity on the main image of index and getting started page
- code snippet formatting

**Comment by dhmlau** [comment]
Thx @b-admike. Fixed it.

File: getting-started.html [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+    </section>
+    <div class="action">
+        <div class="container">
+            <div class="row">
+                <div class="col-md-8">
+                    <h3 style="color:white">Ready to continue your journey?
+                        <p class="p large">Follow the tutorial to create a real-world application with LoopBack 4.</h3>
+                </div>
+                <div class="col-md-4">
+                    <a class="btn btn-primary" role="button" style="padding-left:60px; padding-right:60px" href=https://loopback.io/doc/en/lb4/todo-tutorial.html>View Tutorial</a>
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: confirmed with @shimks , only font color is changed.  I will submit another PR for the hover spec that I got from Cody after his initial file.

File: getting-started.html [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+    <section id="step6" class="even-section"> 
+        <!-- Test your application -->
+        <div class="container">
+            <div class="row text-center">
+                <div class="col-md-12">
+                    <button class="steptag text-uppercase">Step 6</button>
+                </div>
+                <div class="col-md-6">
+                    <h3 class="feature-heading">Test your application</h3>
+                    <p class="p large step">Start the application using <code>npm start</code>.
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: screen shot 2018-08-10 at 5 05 18 pm
hm. should be the same. it's "White-XD"

File: getting-started.html [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+                    </p>
+                </div>
+                <div class="col-md-6">
+                    Visit <a class="link" href="http://127.0.0.1:3000/hello">http://127.0.0.1:3000/hello</a> to see <code>Hello world!</code>
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: screen shot 2018-08-10 at 5 09 19 pm
it should be dark blue but no underline. will remove the underline text decoration.

File: getting-started.html [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+    <section id="step2" class="even-section">
+        <!-- Install LB4 CLI subsection -->
+        <div class="container">
+            <div class="row text-center">
+                <div class="col-md-12">
+                    <button class="steptag text-uppercase">Step 2</button>
+                </div>
+                <div class="col-md-6">
+                    <h3 class="feature-heading">Install LoopBack 4 CLI</h3>
+                    <p class="p large step">The LoopBack 4 CLI is a command-line interface that can scaffold a project or extension. The CLI provides
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: would like to do it in another PR.

File: getting-started.html [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+                    <p class="p large step">Now that we have a basic project created, it’s time to add our own controller. Let’s add a simple “Hello
+                        World” controller as follows.
+                    </p>
+                </div>
+                <div class="col-md-6">
+                    <pre class="code-snippet" style="top:10%">
+<code-blue> $</code-blue> lb4 controller
+<code-blue>[?]</code-blue> Controller class name: hello
+<code-blue>[?]</code-blue> What kind of controller would you like to generate? Empty Controller
+    create src/controllers/hello.controller.ts
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: updated with the new output.

File: getting-started.html [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+                <div class="col-md-12">
+                    <button class="steptag text-uppercase">Step 1</button>
+                </div>
+                <div class="col-md-6">
+                    <h3 class="feature-heading">Have you installed Node.js?</h3>
+                    <p class="p large step">Before you install LoopBack, make sure to download and install Node.js (version 8.9.x or higher), a JavaScript
+                        runtime. </p>
+                </div>
+                <div class="col-md-6">
+                    <a class="btn btn-secondary" style="padding-left:60px; padding-right:60px; top: 10%" role="button" href="https://nodejs.org/en/download/">Install Node.js</a>
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply getting-started.html]: fixed

**Pull request merged**